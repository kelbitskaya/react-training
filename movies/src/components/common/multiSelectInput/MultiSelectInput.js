import React, { useState, useCallback, useMemo } from 'react';
import PropTypes from 'prop-types';
import MultiSelect from 'react-multi-select-component';

export default function MultiSelectInput(props) {
  const { options, defaultValues } = props;
  const defaultValuesArray = defaultValues && defaultValues.split(',') || [];
  const defaultValuesObject = defaultValuesArray.map((item, index) => ({ label: item, value: index }));

  const optionsArr = useMemo(()=> options.map((item, index) => ({ label: item.title, value: index })), [options]);
  const [selected, setSelected] = useState(defaultValuesObject);
  const [values, setValues] =useState('');

  const onChange = useCallback((selectedElements) => {
    setSelected(selectedElements);
    getArrayValues(selectedElements)

  }, []);

  const getArrayValues = (selectedElements) => {
    const resultString = selectedElements.map(item=>item.label).join(', ');
    setValues(resultString)
  };

  return (
    <>
    <MultiSelect
      id="genres"
      options={optionsArr}
      value={selected}
      onChange={onChange}
      hasSelectAll={false}
      selected={selected}
      name='select'
      dataSelect={selected}
    />
      <input type="text" value={values} name="genres" className='hidden-input'/>
    </>
  );
}

MultiSelectInput.propTypes = {
  options: PropTypes.arrayOf(
    PropTypes.shape({
      title: PropTypes.string.isRequired,
    }),
  ).isRequired,
  defaultValues: PropTypes.shape({
    split: PropTypes.func.isRequired,
  }).isRequired,
};
