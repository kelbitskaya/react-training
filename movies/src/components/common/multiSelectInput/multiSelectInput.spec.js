import React from 'react';
import rerender from 'react-test-renderer';
import MultiSelectInput from './MultiSelectInput';
import Constants from '../../constants';

const props = {
  defaultValues: {
    split: ()=>{},
  },
  name: 'select',
  id: 'genres',
  selected: ['Comedy'],
  options: Constants.GENRE,
  selectedElements: [],
  getArrayValues: jest.fn(),
};

describe('MultiSelectInput', () => {
  test('then snapshot created', () => {
    const component = rerender.create(
      <MultiSelectInput
        {...props}
      />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
