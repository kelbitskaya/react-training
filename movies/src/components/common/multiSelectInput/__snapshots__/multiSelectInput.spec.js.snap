// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultiSelectInput then snapshot created 1`] = `
Array [
  <div
    className="go2139093995 multi-select"
  >
    <div
      aria-expanded={false}
      aria-readonly={true}
      className="go2646822163 dropdown-container"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      tabIndex={0}
    >
      <div
        className="go2642161244 dropdown-heading"
        onClick={[Function]}
      >
        <div
          className="dropdown-heading-value"
        >
          <span
            className="gray"
          >
            Select...
          </span>
        </div>
        <svg
          className="dropdown-heading-dropdown-arrow gray"
          fill="none"
          height="24"
          stroke="currentColor"
          strokeWidth="2"
          width="24"
        >
          <path
            d="M6 9L12 15 18 9"
          />
        </svg>
      </div>
    </div>
  </div>,
  <input
    name="genres"
    type="hidden"
    value=""
  />,
]
`;
